src/app/speechbar.service.ts

```diff
@@ -35,18 +38,26 @@ export class SpeechbarService {
       msg.onstart = () => this.listener.next(true);
       msg.onend = () => this.listener.next(false);
       this.speechSynthesizer.speak(msg);
-
     }
   }
 
-  getButtons() {
-    return this.buttons;
+  private notifyButtonObserver() {
+    this.buttonObserver.next(this.buttons);
+  }
+
+  getButtons(): Observable<Button[]> {
+    return new Observable<Button[]>(this.addButtonObserver);
   }
 
   getSpeaking(): Observable<boolean> {
     return new Observable<boolean>(this.addListener);
   }
 
+  addButtonObserver = (observer: Observer<Button[]>) => {
+    this.buttonObserver = observer;
```

Comment:
I assume we need to do the same error handling here when we add it? Around already having an observer?